### PR TITLE
feat: migrate TUI to Textual and add demo subcommand

### PR DESCRIPTION
**Pull request by nh13**
## Summary

- Replaces the 3,300-line `Rich.Live`-based `WorkflowMonitorTUI` with a `SnakeseeApp(textual.App)`. Eliminates the hand-rolled keyboard state machine in favor of declarative `BINDINGS`, focus routing, modal screens, reactives, `DataTable`, `RichLog`, `Footer`, and `Input` widgets. Layout modes (`MINIMAL` / `COMPACT` / `FULL`) are CSS classes.
- Adds `snakesee demo` subcommand: runs a small bundled Snakemake workflow (5 samples, ~26 jobs across 6 rules) under the real Snakemake binary with the snakesee logger plugin writing events. Demo dirs accumulate at `~/.cache/snakesee/demo/` for historical-log nav.

## Architecture

- `snakesee/tui/app.py` — `SnakeseeApp` with bindings, reactives, compose, watchers
- `snakesee/tui/data_source.py` — `WorkflowDataSource` (pure-data layer extracted from the old monitor.py)
- `snakesee/tui/renderables.py` — Rich renderables hosted inside `Static` widgets
- `snakesee/tui/tables.py` — `DataTable` row builders + sort helpers
- `snakesee/tui/screens.py` — `HelpScreen`, `EasterEggScreen`, `JobLogScreen`
- `snakesee/tui/app.tcss` — CSS for layout modes and widget styling
- `snakesee/demo/runner.py` + `snakesee/demo/Snakefile` — demo subcommand machinery

Public API preserved: `from snakesee.tui import WorkflowMonitorTUI` continues to work as an alias for `SnakeseeApp`.

## Test plan

- [ ] `uv run poe check-all` is green (980 tests, mypy strict, ruff format/lint).
- [ ] `uv run snakesee demo --cores 4 --duration short` boots the TUI, renders all six tables against real Snakemake events, dedup[E] failure populates the failed-jobs panel, and `q` quits cleanly.
- [ ] Every keybinding in the spec acts as expected (manual smoke): `tab` cycles layout, `?` shows help, `s`/`S` cycle sort target, `1`–`4` set column, `/` filter + `n`/`N` jump matches, `[`/`]`/`{`/`}` browse historical logs, `f` then `g` opens easter egg, `enter` on a row opens `JobLogScreen`.
- [ ] No `Rich.Live` / `select.select` / `termios` / `tty` code remains in `snakesee/tui/`.

## Summary by CodeRabbit

* **New Features**
  * Bundled demo workflow + new CLI demo command; full Textual-based interactive TUI with multiple layouts, keyboard-driven filtering/sorting/navigation, log viewer, ETA/time‑estimation, accessibility mode, and an easter‑egg.

* **Documentation**
  * Added CLAUDE guidance and updated architecture docs; README now describes a Textual TUI and includes developer guidance for an optional logger plugin.

* **Tests**
  * Large new/expanded test suite covering demo runner, TUI behavior, data source, renderables, tables, and utilities.

* **Chores**
  * Updated packaging and runtime/dev dependencies to support the new TUI and demo.

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

File: tests/test_app_sort.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        assert app.sort_table == "running"
+        await pilot.press("s")
+        assert app.sort_table == "completions"
+        await pilot.press("s")
+        assert app.sort_table == "pending"
+        await pilot.press("s")
+        assert app.sort_table == "stats"
+        await pilot.press("s")
+        assert app.sort_table is None
+        await pilot.press("q")
```
[the remark this answers is not in the document]
Comment (nh13) [reply tests/test_app_sort.py:23]: Verified against current code: `_SORT_CYCLE` is `(None, RUNNING, COMPLETIONS, PENDING, STATS)` — it does not include `failed`/`incomplete`, so the test's forward cycle (wrapping after `stats`) and `None → stats` reverse step match the app's declared cycle. Tests pass as written; no change needed.

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
Re: the `Esc` / `action_clear_filter` suggestion (filter-box-focused branch resetting `current_log_index`) — this is intentional. The behavior is a deliberate two-stage `Esc`: the first `Esc` (filter input focused) just dismisses the in-progress filter edit and preserves your current position, including a historical-log view; a second `Esc` (input no longer focused) clears the applied filter and returns to the latest log. Resetting the log index on the focused branch would yank a user out of a historical-log view merely for cancelling a filter edit, which is more surprising than the staged behavior. The docstring scopes "return to latest log" to the otherwise branch, and the code matches it. Leaving as-is by design.

The stats-table sort issue from the same review is fixed (visible column index now mapped to the correct `StatsRow` field via `sort_stats_rows`, with unit coverage).

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
Addressed the remaining comments from the last review:

- Constructor-supplied `accessibility_config` is now preserved and restored when toggling accessibility off, with a regression test.
- The Thr column sorts numerically (`"10"` after `"8"`), with the `"-"` aggregate row sorting first ascending; the test now includes a 10-thread row plus a sentinel-ordering case.
- The pause guard is now bypassed for **all** explicit user-triggered redraws, not just `r`: pause toggle, `ctrl+r`, estimator re-init (`e`/`w`), sort keys, filter changes, log navigation, and the accessibility toggle. Pausing now gates only the interval timer's automatic polling. Covered by new tests that assert the data source is actually polled while paused.
- `BINDINGS` are now `ClassVar[list[BindingType]]`-annotated rather than converted to tuples: Textual declares `BINDINGS: ClassVar[list[BindingType]]` on `DOMNode`, so a tuple fails mypy strict against the base-class type, and the `ClassVar` annotation is what RUF012 actually asks for.

Re: the suggestion to convert the `tables.py` row models to frozen slots dataclasses — keeping these as `NamedTuple` deliberately. `sort_rows()` is generic over any row type via integer indexing (`TypeVar` bound to `tuple`), which is what lets a single helper sort five different job tables without per-type key maps; `NamedTuple` instances are already immutable; and the rows mirror the data-source tuples with zero conversion cost. The frozen-dataclass convention is aimed at core domain models (`models.py`), not these thin display-row carriers.

The Integration (3.12, 9.14.0) failure on `test_nested_wildcards` was the known event-flush flake (#21) — the re-run passed with no code change.

**Comment by nh13** [comment]
@coderabbitai review
